backend/src/main/java/org/cryptomator/hub/api/UsersResource.java

```diff
@@ -62,6 +67,33 @@ public Response putMe(@Nullable @Valid UserDto dto) {
 		return Response.created(URI.create(".")).build();
 	}
 
+	@POST
+	@Path("/me/access-tokens")
+	@RolesAllowed("user")
+	@Transactional
+	@Consumes(MediaType.APPLICATION_JSON)
+	@Operation(summary = "adds/updates user-specific vault keys", description = "Stores one or more vaultid-vaultkey-tuples for the currently logged-in user, as defined in the request body ({vault1: token1, vault2: token2, ...}).")
+	@APIResponse(responseCode = "200", description = "all keys stored")
+	public Response updateMyAccessTokens(@NotEmpty Map<UUID, String> tokens) {
+		var user = User.<User>findById(jwt.getSubject());
+		for (var entry : tokens.entrySet()) {
+			var vault = Vault.<Vault>findById(entry.getKey());
+			if (vault == null || vault.archived) {
```

Comment:
I'm not quite sure if we want to skip archived vaults during migration. The reason is probably that you don't want to touch it because it's archived, but as far as unlocking is concerned, the migration doesn't change anything, it needs to be migrated later when the vault is unarchived again. I think our goal should be to have as few legacy tokens in the system as possible, or what do you think?

Reply:
I wasn't sure either. Lets decide this in a short call.

Reply:
Decision: We will allow updating access tokens for _every_ accessible vault, regardless of whether it is archived or not. Unlock will of course still be denied for archived vaults, but data migration is in the spirit of the decision that led to archiving (as opposed to deleting): We want to avoid loss of data.